src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -607,14 +607,14 @@ public void PatternTests_UnixEscape(string pattern, string[] sourceFiles, string
         }
 
         [Fact]
-        [PlatformSpecific(TestPlatforms.Windows)]  // Long path segment in search pattern throws PathTooLongException
+        [PlatformSpecific(TestPlatforms.Windows)]  // Long path segment in search pattern throws IOException
         public void WindowsSearchPatternLongSegment()
         {
             // Create a path segment longer than the normal max of 255
             DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
             string longName = new string('k', 257);
 
-            Assert.Throws<PathTooLongException>(() => GetEntries(testDir.FullName, longName));
+            Assert.Throws<IOException>(() => GetEntries(testDir.FullName, longName));
```

Comment:
The underlying error was `Interop.Errors.ERROR_INVALID_PARAMETER` which is currently being mapped to just an IOException. It seemed to generic of an error to convert to PathTooLongException or anything else. That was just my opinion, though. Thoughts?